src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -0,0 +1,263 @@
+        const column = args.column as FilterableColumn<T>;
+        if ((<FilterableColumn<T>>column).filterable === false) {
+            return;
+        }
+        if (args.node.classList.contains("slick-header-with-filter")) {
+            // the the filter button has already being added to the header
+            return;
+        }
+
+        // The default sorting feature is triggered by clicking on the column header, but that is conflicting with query editor grid,
+        // For query editor grid when column header is clicked, the entire column will be selected.
+        // If the column is not defined as sortable because of the above reason, we will add the sort indicator here.
+        if (column.sortable !== true) {
+            args.node.classList.add("slick-header-sortable");
+            append(args.node, $("span.slick-sort-indicator"));
+        }
+
+        args.node.classList.add("slick-header-with-filter");
+        const $el = jQuery(
+            `<button tabindex="-1" id="anchor-btn" aria-label="${ShowFilterText}" title="${ShowFilterText}"></button>`,
+        )
+            .addClass("slick-header-menubutton")
+            .data("column", column);
+        if (column.filterValues?.length) {
+            this.setButtonImage($el, column.filterValues?.length > 0);
+        }
+
+        $el.on("click", async (e: JQuery.ClickEvent) => {
+            e.stopPropagation();
+            e.preventDefault();
+            this.showFilter($el[0]);
+        });
+
+        $el.appendTo(args.node);
+
+        //@ts-ignore
+        this.columnButtonMapping[column.id] = $el[0];
+    }
+
+    private showFilter(filterButton: HTMLElement) {
+        let $menuButton;
+        const target = withNullAsUndefined(filterButton);
+        if (target) {
+            $menuButton = jQuery(target);
+            this.columnDef = $menuButton.data("column");
+        }
+
+        const offset = jQuery(filterButton).offset();
+        if (offset) {
+            // If there is not enough vertical space under the filter button, we will move up the menu.
+            // const menuTop = offset.top + this.menu.offsetHeight <= window.innerHeight ? offset.top : window.innerHeight - this.menu.offsetHeight;
+            // Make sure the menu is on the screen horizontally.
+            // const menuLeft = offset.left + filterButton.offsetWidth + this.menu.offsetWidth <= window.innerWidth ? offset.left + filterButton.offsetWidth : window.innerWidth - this.menu.offsetWidth;
+        }
+
+        var $popup = jQuery(
+            '<div id="popup-menu">' +
+                `<button id="sort-ascending" type="button" icon="slick-header-menuicon.ascending" class="sort-btn">${l10n.t("Sort Ascending")}</button>` +
+                `<button id="sort-descending" type="button" icon="slick-header-menuicon.descending" class="sort-btn">${l10n.t("Sort Descending")}</button>` +
+                `<button id="close-popup" type="button" class="sort-btn">${l10n.t("Cancel")}</button>` +
```

Comment:
Ideally the cancel button shouldn't be placed at the same level as the sorting buttons, but I'll let you decide the best way of displaying the button.

Previously we can click other areas to close the pop-up menu, right?

Reply:
Yes, we can click other areas to close the pop-up menu here too

Reply:
Yeah, we can change the button styling later but I will just merge this in for now